Ray.jl/src/runtime.jl

```diff
@@ -81,7 +91,9 @@ function init(runtime_env::Union{RuntimeEnv,Nothing}=nothing)
     job_config = JobConfig(RuntimeEnvInfo(runtime_env))
     serialized_job_config = _serialize(job_config)
 
-    ray_jll.initialize_driver(args..., job_id, serialized_job_config)
+    log_to_stderr = Base.get(ENV, LOGGING_REDIRECT_STDERR_ENVIRONMENT_VARIABLE, "0") == "1"
+    logs_dir = log_to_stderr ? "" : logs_dir
```

Comment:
so if I'm reading this right...it's not enough to provide a `logs_dir` to get the logs directed to a file...as this can be totally ignored by an environment variable? which seems...convoluted/surprising?

(note: I know this is a Ray thing I'm just confused why the ENV var can obviate the (ostensible) purpose of a function argument...unless the ENV var controls much more than this and there's more going on than I'm aware...)

Reply:
yeah, this might not be a great idea, I was just trying to be consistent with the python implementation (which checks this env var)

Reply:
I think it's better if this option is controllable via arguments from the `Ray.init` call and the `ENV` setting just modifies the default. Something along the lines of:

```julia
function default_log_dir(session_dir)
    redirect_logs = Base.get(ENV, LOGGING_REDIRECT_STDERR_ENVIRONMENT_VARIABLE, "0") == "1"
    return redirect_logs ? "" : joinpath(session_dir, "logs")
end

function init(runtime_env::Union{RuntimeEnv,Nothing}=nothing;
              session_dir="/tmp/ray/session_latest",
              logs_dir=default_log_dir(session_dir))
    ...
end
```

Reply:
The difference with my proposal is that `logs_dir` takes precedence over `ENV["RAY_LOG_TO_STDERR"]` which currently isn't true